### PR TITLE
feature/enabling environment variables unset action

### LINKED ISSUE #4168: Provide a way to unset environment variables from conan
I haven't found any way to unset environment variables in profiles and conan recipes.
Tools module has `environment_append`, but it can only add or replace something, not delete. Empty value is not the same as unbound variable, so `environment_append({"VAR": ""})` does not work in some cases.

`environment_append` could accept `None` values in a dictionary and unset the corresponding variables for them.
For profiles I have no idea yet.

### PR DESCRIPTION
**Pull request by alacasta**
Changelog: Feature: The `tool.environment_append` now accepts unsetting variables by means of appending such variable with a value equal to None. 

Tests covering tools.env.environment_append have been also included. 
Fixes #4168 
Docs: https://github.com/conan-io/docs/pull/1003

**Note**: This is my first contribution to this project so please, apologize if something is not fitting contribution guidelines.

### CODE REVIEW

**Comment by alacasta** [comment]
I've broken CI twice. One of them is because of a different python version used for development (2.7) and the requirements that are also checked (python 3.4). Do you have any Dockerized development environment for Conan containing all the required libraries and tools that Conan should be compatible with?

**Comment by lasote** [comment]
Hi, thanks for the contribution! About the environment, no, we don't have testing environment prepared but we use `tox` that can help to test with different python version. Anyway, don't worry if the CI breaks and you need to push again.

**Comment by alacasta** [comment]
Thanks @lasote! 
Let's see now...

**Comment by jgsogo** [comment]
The previous behavior for `environment_append({'env_var1': None, })` was raising a `TypeError` exception (only `str` values were allowed), so I think it is safe to implement this new _tool_. 
Nevertheless I wonder if `environment_append` is the place for this behavior or we should provide a different `environment_remove` to avoid possible misunderstandings. I just wanted to put this on the table, but I will agree with your decission.

File: conans/client/tools/env.py [hunk 1]
```diff
@@ -46,6 +53,8 @@ def environment_append(env_vars):
     if env_vars:
         old_env = dict(os.environ)
         os.environ.update(env_vars)
+        for var in unset_vars:
```
Comment (jgsogo) [inline conans/client/tools/env.py]: Execution won't get here if `env_vars` is empty (if all the variables are `=None`). Please, add a test case for this.

**Comment by lasote** [comment]
I suggested a different tool at some point but finally, I think keeping it in the same could be easier and for me is also easy to understand as a user. I would vote for implementing it in the `environment_append`

**Comment by alacasta** [comment]
Thanks for your comments @jgsogo!

Regarding
> The previous behavior for `environment_append({'env_var1': None, })` was raising a `TypeError` exception (only `str` values were allowed), so I think it is safe to implement this new _tool_.
> Nevertheless I wonder if `environment_append` is the place for this behavior or we should provide a different `environment_remove` to avoid possible misunderstandings. I just wanted to put this on the table, but I will agree with your decission.

I must say that at the beginning i was from your opinion. However and after carefuly analyzing all the points discussed on #4168 seems like having a different tool for this purpose looks complementary to the use case suggested there (more oriented to usetting variables via profiles). Anyway, if this feature is finally included, this change will become automatically unused and will be able to be refactored. 

We can just keep an eye on this if this happen :)

File: conans/client/tools/env.py (hunk 1 above)
Comment (alacasta) [reply conans/client/tools/env.py]: Good catch!!

Absolutely right. Let me fix & create a regression test case!

**Comment by kdsx** [comment]
Thanks for the fix!
By the way, does it work with profiles now?

**Comment by jgsogo** [comment]
@kdsx, no, this PR does nothing related to the profiles. As commented (https://github.com/conan-io/conan/issues/4168#issuecomment-450802939) the profiles should be a different feature, is there any issue opened specific to the profiles implementation?

**Comment by kdsx** [comment]
No, I haven't created it.

**Pull request merged**